### PR TITLE
UrlJob proxy support

### PR DESCRIPTION
**Pull request by lechuckcaptain**
Added proxy support to UrlJob

### CODE REVIEW

**Comment by thp** [comment]
Does this break using environment variables for proxy support? Other than that, looks good.

**Comment by lechuckcaptain** [comment]
You're right! I added the read of proxy settings from os environment. The system settings can be overriden (or directly defined when is not possible to set environment variables) by the configuration added into urls.yaml file.

File: lib/urlwatch/jobs.py [hunk 1]
```diff
@@ -26,8 +26,7 @@
 # THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
 # (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF
 # THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
-
```
Comment (thp) [inline lib/urlwatch/jobs.py]: Please leave in the newlines here, looks much better.

File: lib/urlwatch/jobs.py [hunk 2]
```diff
@@ -182,18 +181,24 @@ class UrlJob(Job):
...
-    __optional__ = ('cookies', 'data', 'method', 'ssl_no_verify')
+    __optional__ = ('cookies', 'data', 'method', 'ssl_no_verify', 'http_proxy', 'https_proxy')
 
     CHARSET_RE = re.compile('text/(html|plain); charset=([^;]*)')
 
     def get_location(self):
         return self.url
 
     def retrieve(self, job_state):
+
```
Comment (thp) [inline lib/urlwatch/jobs.py]: Please no empty newlines at the start of function body.

File: lib/urlwatch/jobs.py [hunk 3]
```diff
@@ -182,18 +181,24 @@ class UrlJob(Job):
...
         headers = {
             'User-agent': urlwatch.__user_agent__,
         }
 
+        proxies = {
+            'http': os.getenv('HTTP_PROXY'),
+            'https': os.getenv('HTTPS_PROXY')
```
Comment (thp) [inline lib/urlwatch/jobs.py]: Add trailing `,` at the end of this line, it's easier to add new items to the dict this way in the future if needed.

**Comment by lechuckcaptain** [comment]
Applied feedback from @thp. Should be fine now.

**Comment by thp** [comment]
All good now, thanks :)

**Pull request merged**